packages/react-component-library/src/components/NumberInputE/useEarlyValidation.ts

```diff
@@ -1,15 +1,44 @@
 import React, { useCallback } from 'react'
 
-import { areCharactersValid } from './validation'
+import { isValueValid } from './validation'
 
-export function useEarlyValidation(): {
-  handleKeyDown: (event: React.KeyboardEvent<HTMLInputElement>) => void
+/**
+ * Given some text that is being entered in an input, work out what the
+ * new value will be based on the cursor position ond selection.
+ */
+function getNewInputValue(input: HTMLInputElement, newText: string): string {
```

Comment:
`handleBeforeInput` uses this now so it can check the entire string instead of just the new characters (this enables it to check that the minus sign is at the beginning of the string).